receiver/bigipreceiver/internal/models/nodes.go

```diff
@@ -0,0 +1,61 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver/internal/models"
+
+// Nodes represents the top level json returned by the node/stats endpoint
+type Nodes struct {
+	Entries map[string]NodeStats `json:"entries"`
+}
+
+// NodeStats represents the statistics returned for a single node
+type NodeStats struct {
+	NestedStats struct {
+		Entries struct {
+			Name struct {
+				Description string `json:"description,omitempty"`
+			} `json:"tmName,omitempty"`
+			IPAddress struct {
+				Description string `json:"description,omitempty"`
+			} `json:"addr,omitempty"`
+			ServersideBitsIn struct {
+				Value int64 `json:"value"`
+			} `json:"serverside.bitsIn,omitempty"`
+			ServersideBitsOut struct {
+				Value int64 `json:"value"`
+			} `json:"serverside.bitsOut,omitempty"`
+			ServersideCurConns struct {
+				Value int64 `json:"value"`
+			} `json:"serverside.curConns,omitempty"`
+			ServersidePktsIn struct {
+				Value int64 `json:"value"`
+			} `json:"serverside.pktsIn,omitempty"`
+			ServersidePktsOut struct {
+				Value int64 `json:"value"`
+			} `json:"serverside.pktsOut,omitempty"`
+			AvailabilityState struct {
+				Description string `json:"description,omitempty"`
+			} `json:"status.availabilityState,omitempty"`
+			EnabledState struct {
+				Description string `json:"description,omitempty"`
+			} `json:"status.enabledState,omitempty"`
+			TotalRequests struct {
+				Value int64 `json:"value"`
+			} `json:"totRequests,omitempty"`
+			CurSessions struct {
+				Value int64 `json:"value"`
+			} `json:"curSessions,omitempty"`
+		} `json:"entries,omitempty"`
+	} `json:"nestedStats,omitempty"`
+}
```

Comment:
This is something that's been brought up in the past when using this approach (using a struct for unmarshalling payloads), but are these fields all guaranteed to be there? The main downside of this approach is that you can't tell the difference between the zero value and the absence of the field.

As long as we're confident that big-ip always return values for the fields, it should be good, but just thought I'd ask.

Reply:
If we're not confident something will be there we can make it a pointer. Pointer zero value is `nil` so it's easier to check if something exists or not.

Reply:
@BinaryFissionGames i feel fairly confident that they should always return these fields.